encoders/firebase-decoders-json/src/main/java/com/google/firebase/decoders/TypeToken.java

```diff
@@ -80,8 +86,8 @@
       }
       throw new IllegalArgumentException("<? super T> is not supported");
     } else if (type instanceof GenericArrayType) {
-      Type componentType = ((GenericArrayType) type).getGenericComponentType();
-      return new ArrayToken<T>(TypeToken.of(componentType));
+      // TODO: Support GenericArrayType
+      throw new IllegalArgumentException("GenericArrayType is not supported.");
```

Comment:
I've read a bit more on generic arrays and it looks like they are even more weird than I'd thought, i.e. there is not even a "sane" way to create one - it just won't compile - even though the type `Optional<String>[]` itself is legal in java.

So I would suggest you remove the todo and just leave it as explicitly not supported.

Also consider using the error message rtdb uses https://github.com/firebase/firebase-android-sdk/blob/a7096ac2a65842f9528243b2451e5b172de4c789/firebase-database/src/main/java/com/google/firebase/database/core/utilities/encoding/CustomClassMapper.java#L200